spinalcordtoolbox/scripts/sct_deepseg.py

```diff
@@ -130,6 +132,48 @@ def get_parser():
     return parser
 
 
+def segment_volume_with_ensemble(model_paths, input_filenames, options):
+    """
+        Run `ivadomed.inference.segment_volume()` once per model, then average the outputs.
+
+        :param model_paths: A list of folder paths. The folders must contain:
+            (1) the model ('folder_model/folder_model.pt') to use
+            (2) its configuration file ('folder_model/folder_model.json') used for the training,
+            see https://github.com/neuropoly/ivadomed/wiki/configuration-file
+        :param input_filenames: list of image filenames (e.g. .nii.gz) to segment. Multichannel models require multiple
+            images to segment, i.e.., len(fname_images) > 1.
+        :param options: A dictionary containing optional configuration settings, as specified by the
+            ivadomed.inference.segment_volume function.
+
+        :return: list, list: List of nibabel objects containing the soft segmentation(s), one per prediction class, \
+            List of target suffix associated with each prediction
+    """
+    # Fetch the name of the model (to be used in logging)
+    name_model = Path(model_paths[0]).parts[-1]
+    # Perform inference once per model in the ensemble
+    nii_lsts, target_lsts = [], []
+    for path_model in model_paths:
+        name_seed = Path(path_model).parts[-2]
+        logger.info(f"\nRunning inference using model '{name_model}' ({name_seed})...")
+        nii_lst, target_lst = imed_inference.segment_volume(path_model, input_filenames, options=options)
+        nii_lsts.append(nii_lst)
+        target_lsts.append(target_lst)
+    # The 'niis' are images, so average the image data across the ensemble
+    logger.info(f"\nAveraging outputs across the ensemble for '{name_model}'...")
+    nii_lst = []
+    n_outputs_per_model = len(nii_lsts[0])
+    for i in range(n_outputs_per_model):
+        # Average the data for each output in the ensemble
+        data_mean = np.mean([im_lst[i].get_fdata() for im_lst in nii_lsts])
+        # Take the first image's header to reuse for the averaged image
+        nii_header = nii_lsts[0][i].header
+        # Create a new Nifti1Image containing the averaged output
+        nii_lst.append(nib.Nifti1Image(data_mean, header=nii_header, affine=nii_header.get_best_affine()))
```

Comment:
Nitpick: you can iterate over the thing itself rather than the indices here, if you want to.
```suggestion
    for predictions in zip(*nii_lsts):
        # Average the data for each output in the ensemble
        data_mean = np.mean([pred.get_fdata() for pred in predictions])
        # Take the first image's header to reuse for the averaged image
        nii_header = predictions[0].header
        # Create a new Nifti1Image containing the averaged output
        nii_lst.append(nib.Nifti1Image(data_mean, header=nii_header, affine=nii_header.get_best_affine()))
```

Reply:
!!! This is wonderful! Thank you so much. :hearts: 

I couldn't figure out how to iterate over the inner lists in parallel, but `zip(*nii_lsts)` is a perfect way to do it.

Reply:
Added in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4042/commits/1dcc5f2f8d1fd25d1bab821dac1a558c90f070b9.